azure-ipam/logger/logger.go

```diff
@@ -6,56 +6,48 @@ import (
 	"github.com/pkg/errors"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
 )
 
 type Config struct {
-	Level            string
-	OutputPaths      string // comma separated list of paths
-	ErrorOutputPaths string // comma separated list of paths
+	Level       string // Debug by default
+	Filename   	string
+	MaxSize    	int    // megabytes
+	MaxBackups 	int    // # of backups, no limitation by default
+	MaxAge     	int    // days, no limitation by default
 }
 
 // NewLogger creates and returns a zap logger and a clean up function
 func New(cfg *Config) (*zap.Logger, func(), error) {
-	loggerCfg := &zap.Config{}
-
-	level, err := zapcore.ParseLevel(cfg.Level)
-	if err != nil {
-		return nil, nil, errors.Wrapf(err, "failed to parse log level")
-	}
-	loggerCfg.Level = zap.NewAtomicLevelAt(level)
-
-	loggerCfg.Encoding = "json"
-	loggerCfg.OutputPaths = getLogOutputPath(cfg.OutputPaths)
-	loggerCfg.ErrorOutputPaths = getErrOutputPath(cfg.ErrorOutputPaths)
-	loggerCfg.EncoderConfig = zapcore.EncoderConfig{
-		TimeKey:     "time",
-		MessageKey:  "msg",
-		LevelKey:    "level",
-		EncodeLevel: zapcore.LowercaseLevelEncoder,
-		EncodeTime:  zapcore.ISO8601TimeEncoder,
-	}
-
-	logger, err := loggerCfg.Build()
-	if err != nil {
+	//check the filepath is not empty
+	if cfg.Filename == "" {
+		err := errors.New("no Filename is provided")
```

Comment:
no filename should not be an error, it just means don't log to a file

Reply:
Lumberjack log in <processname>-lumberjack.log within os.TempDir() if filename is empty. 

I decided to avoid such scenario. If we make this package universal, then we can address this better.

Reply:
I think you misunderstood what I meant - if I don't specify a log file, you should not log to a file at all